pkg/internal/templates.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+const wrappedInterfacesTempl = `
+
+//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
+
+// Code auto-generated. DO NOT EDIT.
+
+package {{.ClientsetName}}
+
+import (
+	"fmt"
+	
+	kcp "github.com/kcp-dev/apimachinery/pkg/client"
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	{{$name := .InputPath}}
+	{{ range .APIs }}
+	{{.PkgName}}{{.Version}} "{{$name}}/pkg/apis/{{.PkgName}}/{{.Version}}"
+	{{ end }}
+)
+
+func NewForConfig(config *rest.Config) (*ClusterClient, error) {
+	client, err := rest.HTTPClientFor(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating HTTP client: %w", err)
+	}
+
+	clusterRoundTripper := kcp.NewClusterRoundTripper(client.Transport)
+	client.Transport = clusterRoundTripper
+
+	delegate, err := kubernetes.NewForConfigAndClient(config, client)
+	if err != nil {
+		return nil, fmt.Errorf("error creating delegate clientset: %w", err)
+	}
+
+	return &ClusterClient{
+		delegate: delegate,
+	}, nil
+}
+
+type ClusterClient struct {
+	delegate {{.InterfaceName}}.Interface
+}
+
+func (c *ClusterClient) Cluster(cluster logicalcluster.LogicalCluster) {{.InterfaceName}}.Interface {
+	return &wrappedInterface{
+		cluster:  cluster,
+		delegate: c.delegate,
+	}
+}
+
+type wrappedInterface struct {
+	cluster  logicalcluster.LogicalCluster
+	delegate {{.InterfaceName}}.Interface
```

Comment:
Instead of InterfaceName, this needs to be the package that has the clientset you're wrapping, e.g. `github.com/kcp-dev/kcp/pkg/client/clientset`.